docs/workflow-and-job.md

```diff
@@ -10,151 +10,222 @@ It can execute anything that can be executed from the Python code, for example:
-It depends on the `Workflow` setup. By default, it will be date only - `YYYY-MM-DD`.
+## Workflow
 
-When you run a single job through the workflow class, without providing the `runtime` parameter, the `Workflow` class
-will pass the current date time as default.
+The `Workflow` class takes 2 main parameters - `workflow_id` and `definition`.
 
-```python
-# Let's assume that now == '2020-01-02 01:11:00'
-simple_workflow = Workflow(workflow_id='simple_workflow', definition=[SimpleJob('1'), SimpleJob('2')])
-hello_world_workflow.run_job('2')
->>> '2020-01-02 01:11:00'
-```
+The `workflow_id` parameter is a string that should be a valid Python variable name. For example - `'my_example_workflow'`, `'MY_EXAMPLE_WORKFLOW'`, `'workflow1234'`.
 
-There are 2 additional parameters, that job can supply - `retry_count` and `retry_pause_sec`. The `retry_count` parameter
-determines how many times a job will be retried. The `retry_pause_sec` sets how long the pause between retries should be.
+The `Workflow` class arranges jobs into a DAG, through the `definition` parameter. There are 2 ways of specifying job arrangement. First one is passing a list
+of jobs:
 
+[`./docs_examples/workflow_and_job/sequential_workflow.py`](./docs_examples/workflow_and_job/sequential_workflow.py)
 ```python
-class SimpleJob:
+from bigflow.workflow import Workflow
+
+class Job(object):
     def __init__(self, id):
         self.id = id
-        self.retry_count = 20
-        self.retry_pause_sec = 100
 
     def run(self, runtime):
-        print(runtime)
+        print(f'Running job {self.id} at {runtime}')
+
+example_workflow = Workflow(
+    workflow_id='example_workflow',
+    definition=[Job('1'), Job('2')])
 
-job = SimpleJob('my_simple_job')
+example_workflow.run()
 ```
 
-## Workflow
+Output:
+```text
+Running job 1 at 2020-01-01
+Running job 2 at 2020-01-01
+```
 
-The `Workflow` class arranges jobs into a DAG. There are 2 ways of specifying job arrangement. First one is passing a list
-of jobs to `Workflow`:
+The second one is passing the `Definition` object, which allows you to create a graph.
 
+Let us say that we want to create the following DAG:
+    
+```
+       |--job2--|
+job1-->          -->job4 
+       |--job3--|
+```
+
+The implementation looks like this:
+
+[`./docs_examples/workflow_and_job/graph_workflow.py`](./docs_examples/workflow_and_job/graph_workflow.py)
 ```python
-from biggerquery.build import Workflow
+job1, job2, job3, job4 = Job('1'), Job('2'), Job('3'), Job('4')
 
-class Job(object):
-    def __init__(self, id):
-        self.id = id
-    
-    def run(self, runtime):
-        print(runtime)
+graph_workflow = Workflow(workflow_id='graph_workflow', definition=Definition({
+    job1: (job2, job3),
+    job2: (job4,),
+    job3: (job4,)
+}))
+graph_workflow.run()
+```
 
-example_workflow = Workflow(definition=[Job('1'), Job('2')])
+Output:
+```text
+Running job 1 at 2020-01-01
+Running job 2 at 2020-01-01
```

Comment:
is this Output worth to be written here?

Reply:
done